### PR TITLE
FORG-30 FORG-27 BasicAttacker + BasicEnemy

### PR DESCRIPTION
**Pull request by vakor03**
https://froggies-studio.atlassian.net/browse/FORG-27
https://froggies-studio.atlassian.net/browse/FORG-30

### CODE REVIEW

File: Frogies-Project/Assets/Scripts/Core/Entities/Enemies/BasicEnemy.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            var animationController = new PlayerAnimationController(data.AnimationStateManager, data.SpriteFlipper);
+
+            Brain = new EntityBrain(data.MovementData, data.AttacksData, _inputMoveProvider,
+                _inputFightingInputProvider,
+                data.DirectionalMover, animationController, data.StatsStorage, data.AttackColliders);
+            
+            data.DamageReceiver.Initialize(Brain.HealthSystem.TakeDamage);
+        }
+
+        private bool IsInAttackRange
```
Comment (Sorpirit) [inline Frogies-Project/Assets/Scripts/Core/Entities/Enemies/BasicEnemy.cs:41]: Ordering, please move above private varaibles

File: Frogies-Project/Assets/Scripts/Core/Entities/Player/Player.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+﻿namespace Core.Entities.Player
+{
+    public class Player : BasicEntity
```
Comment (Sorpirit) [inline Frogies-Project/Assets/Scripts/Core/Entities/Player/Player.cs:3]: hehe what a lovely wrapper) feels wrong a bit, but I guess it does make sense

File: Frogies-Project/Assets/Animation/Player/PF Village Props - Banner.controller
Comment (Sorpirit) [inline Frogies-Project/Assets/Animation/Player/PF Village Props - Banner.controller:1]: animation controller on the banner huh?

**Review by Sorpirit — CHANGES_REQUESTED** [review]
Looks good to me, just a few points

File: Frogies-Project/Assets/Animation/Player/PF Village Props - Banner.controller
Comment (vakor03) [reply Frogies-Project/Assets/Animation/Player/PF Village Props - Banner.controller:1]: Lol, idk

**Pull request merged**